src/DynamoCore/Configuration/PreferenceSettings.cs

```diff
@@ -48,12 +48,14 @@ internal static void CopyProperties(this PreferenceSettings source, PreferenceSe
     /// </summary>
     public class PreferenceSettings : NotificationObject, IPreferences, IRenderPrecisionPreference, IDisablePackageLoadingPreferences, ILogSource, IHideAutocompleteMethodOptions
     {
+        internal static PreferenceSettings dynamoModelRuntimePreferenceSettings;
+
         internal readonly static Lazy<PreferenceSettings>
             lazy = new Lazy<PreferenceSettings>
             (() => PreferenceSettings.Load(PathManager.Instance.PreferenceFilePath));
 
         [XmlIgnore]
-        public static PreferenceSettings Instance { get { return lazy.Value; } }
+        public static PreferenceSettings Instance { get { return dynamoModelRuntimePreferenceSettings ?? lazy.Value; } }
 
```

Comment:
so now Dynamo supports local instances of Preferences, a singleton instance which can be loaded from disk, or set to incoming local Preferences instances during the DynamoModel construction (not all the time though).
We need a followup to untangle all of this before it gets out of hand.

At the moment I would not be able to predict what instance of the Preferences class would be used at a given spot in Dynamo.

Reply:
As I understand the behavior today on first time start-up (ie no existing preference XML)
SplashScreen calls `PreferenceSettings.Instance`
`Instance` creates a new` PreferenceSettings` object.
`DynamoModel` start is passed the new `PreferenceSettings` object.
`DynamoModel` performs additional initialization on the new `PrefenceSettings` object
Eventually `DynamoModel` shutdown saves the `PreferenceSettings` to disk

Reply:
Maybe the theme of this PR is looking at all the times Dynamo Player is using `PreferenceSettings.Instance` AKA DynamoPreferences singleton we were expecting to have the Runtime Preferences object.

Reply:
I can give it a shot.  I believe one a `lazy` value is set it doesn't recall the into function over and over.

Reply:
I think the main issue will be all the special handling we do for testing but let's see

Reply:
Also we currently don't let anyone just set the DynamoModel.Prefence it is readonly

Reply:
Which is why we can keep it for the time being if we're worried about API breaks and simply reroute it to `PreferenceSettings.Instance` in its getter:
```
[Obsolete("this will be removed in 4.0")]
public readonly PreferenceSettings PreferenceSettings { get {return PreferenceSettings.Instance;} }
```